hazelcast/src/main/java/com/hazelcast/util/executor/LoggingScheduledExecutor.java

```diff
@@ -50,28 +50,49 @@
  */
 public class LoggingScheduledExecutor extends ScheduledThreadPoolExecutor {
 
+    private static final int DEFAULT_PURGE_INITIAL_DELAY = 0;
+    private static final int DEFAULT_PURGE_PERIOD = 10;
+    private static final TimeUnit DEFAULT_PURGE_TIME_UNIT = TimeUnit.SECONDS;
+
     private final ILogger logger;
     private volatile boolean shutdownInitiated;
 
     public LoggingScheduledExecutor(ILogger logger, int corePoolSize, ThreadFactory threadFactory) {
         super(corePoolSize, threadFactory);
         this.logger = checkNotNull(logger, "logger cannot be null");
+        enablePeriodicPurge();
     }
 
     public LoggingScheduledExecutor(ILogger logger, int corePoolSize, ThreadFactory threadFactory,
                                     RejectedExecutionHandler handler) {
         super(corePoolSize, threadFactory, handler);
         this.logger = checkNotNull(logger, "logger cannot be null");
+        enablePeriodicPurge();
```

Comment:
can we start this purge-task in `cancel` method after first call of it?

Reply:
We can, but I see no point. It will reduce clarity with no big benefit.

Reply:
sure not crucial but the point was to call it when needed, using less resource